portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,44 +33,56 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some placeholder comment saying 'Hello, Omar!'. TODO(oumontiel): modify this file to handle comments data */
+/** Servlet that posts and retrieves comments from Datastore. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   /**
-   * List of messages to be fetched from this Data Servlet
-   * @private {ArrayList<String>}
+   * Gets the comments, in JSON format, taken from user input in the form from the HTML,
+   * which are stored in the messages variable.
    */
-  private ArrayList<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Once upon a time...");
-    messages.add("Twice upon two times...");
-    messages.add("Thrice upon three times...");
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //Create a query and prepare it with the data stored in Datastore.
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    //Add all queried comments from Datastore
+    //to an List of type Comment.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String content = (String) entity.getProperty("content");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, content, timestamp);
+      comments.add(comment);
+    }
+    
+    //Convert the comments List to JSON format.
```

Comment:
I don't see this in your next PR (which I'm reviewing right now), but just remember to add a space between the '//' and the rest of the comment for single-line comments like these! https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

Reply:
Noted! I'll modify this in the last branch (which is week-3/delete).